### PR TITLE
Use the new python module building framework

### PR DESCRIPTION
**Pull request by brianjmurrell**
Signed-off-by: Brian J. Murrell <brian.murrell@intel.com>

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -17,21 +19,21 @@
...
-    include_package_data=True,
-    author='IML Team',
-    url='https://pypi.python.org/pypi/iml-common',
-    description='Common library used by multiple IML components',
-    long_description=long_description,
-    classifiers=[
+    name = 'iml-common',
+    version = package_version(),
+    author = "Intel Corporation",
+    author_email = "hpdd-info@intel.com",
```
Comment (jgrund) [inline setup.py]: iml@intel.com

File: include/copr.mk [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+#copr_build:
+# https://pagure.io/copr/copr/issue/207
+copr_build iml_copr_build: $(PREREQ)
+	# buildpypi is pretty useless right now:
+	# https://pagure.io/copr/copr/issue/207
+	#copr-cli buildpypi --packagename $(NAME)
+	#$(COPR_OWNER)/$(COPR_PROJECT)
+	copr-cli $(COPR_CONFIG) build $(OWNER_PROJECT) $^
+else
+copr_build iml_copr_build: $(RPM_SPEC)
```
Comment (johnsonw) [inline include/copr.mk]: Does this need the `Registry` case as well?

File: python-iml-common.spec [hunk 3]
```diff
@@ -0,0 +1,100 @@
+# Created by pyp2rpm-3.2.2
+%global pypi_name iml-common
+%{!?name: %global name python2-%{pypi_name}}
+%{?!version: %global version 1.3.3}
```
Comment (jgrund) [inline python-iml-common.spec]: Just a note that you will want to cut a new backwards incompatible version, the changes I made prior will not work correctly in IML < 4.1.0

File: include/copr.mk (hunk 2 above)
Comment (brianjmurrell) [reply include/copr.mk]: Yes it should.  Pulled in from module-tools.

File: .travis.yml [hunk 4]
```diff
@@ -1,11 +1,43 @@
...
+    env: Type='mock build test'
+    script:
+    - ./include/travis/run_in_centos7_docker.sh include/travis/mock_build_test.sh
+  - stage: deploy-pypi
+    if: branch =~ ^v\d+\.\d+\.\d+-1.*$
+    env: Type='PyPi deploy'
+    script: skip
+    deploy:
+      provider: pypi
+      user: iml
```
Comment (jgrund) [inline .travis.yml]: Going forward, we should make this a travis env var so we don't expose usernames.

Comment (brianjmurrell) [reply .travis.yml]: Agreed.  Just to note, we are doing this for our other current pypi module(s) so this change needs to be made elsewhere too (looking at you iml_sos_plugin).

**Pull request merged**